src/plugins/formulas/formulas.js

```diff
@@ -88,356 +38,297 @@ export class Formulas extends BasePlugin {
-    this.addHook('beforeRemoveCol', (...args) => this.onBeforeRemoveCol(...args));
-    this.addHook('beforeRemoveRow', (...args) => this.onBeforeRemoveRow(...args));
-    this.addHook('beforeValidate', (...args) => this.onBeforeValidate(...args));
-    this.addHook('beforeValueRender', (...args) => this.onBeforeValueRender(...args));
-    this.addHook('modifyData', (...args) => this.onModifyData(...args));
+    /**
+     * The HyperFormula instance that will be used for this instance of Handsontable.
+     *
+     * @type {HyperFormula}
+     */
+    this.hyperformula = HyperFormula.buildEmpty({
+      licenseKey: 'non-commercial-and-evaluation' // TODO
+    });
 
-    this.sheet.addLocalHook('afterRecalculate', (...args) => this.onSheetAfterRecalculate(...args));
+    this.sheetName = this.hyperformula.addSheet();
 
-    super.enablePlugin();
-  }
+    this.addHook('afterLoadData', (...args) => this.onAfterLoadData(...args));
+    this.addHook('modifyData', (...args) => this.onModifyData(...args));
+    this.addHook('modifySourceData', (...args) => this.onModifySourceData(...args));
 
-  /**
-   * Disables the plugin functionality for this Handsontable instance.
-   */
-  disablePlugin() {
-    super.disablePlugin();
-  }
+    // TODO test if the `before` hook will actually block operations
+    this.addHook('beforeCreateRow', (...args) => this.onBeforeCreateRow(...args));
+    this.addHook('beforeCreateCol', (...args) => this.onBeforeCreateCol(...args));
 
-  /**
-   * Returns cell value (evaluated from formula expression) at specified cell coords.
-   *
-   * @param {number} row Row index.
-   * @param {number} column Column index.
-   * @returns {*}
-   */
-  getCellValue(row, column) {
-    const cell = this.sheet.getCellAt(row, column);
+    this.addHook('afterCreateRow', (...args) => this.onAfterCreateRow(...args));
+    this.addHook('afterCreateCol', (...args) => this.onAfterCreateCol(...args));
 
-    return cell ? (cell.getError() || cell.getValue()) : void 0;
-  }
+    this.addHook('beforeRemoveRow', (...args) => this.onBeforeRemoveRow(...args));
+    this.addHook('beforeRemoveCol', (...args) => this.onBeforeRemoveCol(...args));
 
-  /**
-   * Checks if there are any formula evaluations made under specific cell coords.
-   *
-   * @param {number} row Row index.
-   * @param {number} column Column index.
-   * @returns {boolean}
-   */
-  hasComputedCellValue(row, column) {
-    return this.sheet.getCellAt(row, column) !== null;
-  }
+    this.addHook('afterRemoveRow', (...args) => this.onAfterRemoveRow(...args));
+    this.addHook('afterRemoveCol', (...args) => this.onAfterRemoveCol(...args));
 
-  /**
-   * Recalculates all formulas (an algorithm will choose the best method of calculation).
-   */
-  recalculate() {
-    this.sheet.recalculate();
-  }
+    // Autofill hooks
+    {
+      // Scoped into this block instead of being on the whole class to prevent
```

Comment:
This part really stretches the `enablePlugin` method, and imho makes the whole file much less readable - what do you think about moving the autofill logic to a separate file?

Reply:
https://github.com/handsontable/handsontable/pull/7648/commits/dcbaad93810287fe9b93b30ec3be107d703d6557